configcatclienttests/test_lazyloadingcachepolicy.py

```diff
@@ -58,6 +61,33 @@ def test_force_refresh(self):
         self.assertEqual(config_fetcher.get_call_count, 2)
         cache_policy.stop()
 
+    def test_force_refresh_honours_304(self):
+        config_fetcher = mock.MagicMock()
+        successful_response = mock.MagicMock()
+        successful_response.status_code = 200
+        successful_response.json.return_value = TEST_JSON
+        not_modified_response = mock.MagicMock()
+        not_modified_response.status_code = 304
+        not_modified_response.json.side_effect = ValueError("this response contains no body")
+        config_fetcher.get_configuration_json.return_value = successful_response
+        config_cache = InMemoryConfigCache()
+        cache_policy = LazyLoadingCachePolicy(config_fetcher, config_cache, 160)
+
+        # Get value from Config Store, which indicates a config_fetcher call
+        with mock.patch('configcatclient.lazyloadingcachepolicy.datetime') as mock_datetime:
+            mock_datetime.datetime.utcnow.return_value = datetime.datetime(2020, 5, 20, 0, 0, 0)
+            value = cache_policy.get()
+            self.assertEqual(mock_datetime.datetime.utcnow.call_count, 2)
+            self.assertEqual(value, TEST_JSON)
+            config_fetcher.get_configuration_json.return_value = not_modified_response
+            cache_policy.force_refresh()
+            self.assertEqual(value, TEST_JSON)
+            self.assertEqual(config_fetcher.get_configuration_json.call_count, 2)
+            # this indicates that is_fetched() was correctly called and
+            # the setting of the new last updated didn't occur
+            self.assertEqual(mock_datetime.datetime.utcnow.call_count, 2)
+        cache_policy.stop()
+
```

Comment:
```suggestion
    def test_force_refresh_not_modified_config(self):
        config_fetcher = mock.MagicMock()
        successful_fetch_response = mock.MagicMock()
        successful_fetch_response.is_fetched.return_value = True
        successful_fetch_response.json.return_value = TEST_JSON
        not_modified_fetch_response = mock.MagicMock()
        not_modified_fetch_response.is_fetched.return_value = False
        config_fetcher.get_configuration_json.return_value = successful_fetch_response
        config_cache = InMemoryConfigCache()
        cache_policy = LazyLoadingCachePolicy(config_fetcher, config_cache, 160)

        # Get value from Config Store, which indicates a config_fetcher call
        with mock.patch('configcatclient.lazyloadingcachepolicy.datetime') as mock_datetime:
            mock_datetime.datetime.utcnow.return_value = datetime.datetime(2020, 5, 20, 0, 0, 0)
            value = cache_policy.get()
            self.assertEqual(mock_datetime.datetime.utcnow.call_count, 2)
            self.assertEqual(value, TEST_JSON)
            config_fetcher.get_configuration_json.return_value = not_modified_fetch_response
            cache_policy.force_refresh()
            self.assertEqual(config_fetcher.get_configuration_json.call_count, 2)
            # this indicates that is_fetched() was correctly called and
            # the setting of the new last updated didn't occur
            self.assertEqual(mock_datetime.datetime.utcnow.call_count, 2)
        cache_policy.stop()
```

Reply:
Hey tyroneabdy,
Thanks for the fix so much. I suggested some changes regarding to the test.
If you agree with the changes you can apply it to the PR directly from my comment.

Reply:
Hey @kp-cat I've made some changes to the test that I forgot to push, it was slightly wrong. It wasn't doing exactly what I thought, but I've updated that and included these test changes you've suggested.